tests/trace_server/test_threads_query_builder.py

```diff
@@ -761,6 +761,360 @@ def test_turn_filtering_explanation():
+            -1 as p99_turn_duration_ms
+        FROM calls c1
+        WHERE project_id = ?
+            AND thread_id IS NOT NULL
+            AND thread_id != ''
+            AND id = turn_id                 -- Only include turn calls for meaningful thread stats
+
+            AND thread_id = ?
+        GROUP BY thread_id
+        ORDER BY last_updated DESC
+        """,
+        ["test_project", "another_thread_id"],
+        project_id="test_project",
+        thread_id="another_thread_id",
+    )
+
+
+def test_clickhouse_with_thread_id_and_date_filters():
+    """Test ClickHouse query with both thread_id and date filters."""
+    after_date = datetime.datetime(2024, 1, 1, 12, 0, 0)
+    before_date = datetime.datetime(2024, 12, 31, 23, 59, 59)
+
+    assert_clickhouse_sql(
+        """
+        SELECT
+            thread_id,
+            COUNT(*) as turn_count,
+            min(call_start_time) as start_time,          -- Earliest start time across all calls in thread
+            max(call_end_time) as last_updated,          -- Latest end time across all calls in thread
+            argMin(id, call_start_time) as first_turn_id,     -- Turn ID with earliest start_time
+            argMax(id, call_end_time) as last_turn_id,   -- Turn ID with latest last_updated
+            quantile(0.5)(call_duration) as p50_turn_duration_ms,  -- P50 of turn durations in milliseconds
+            quantile(0.99)(call_duration) as p99_turn_duration_ms  -- P99 of turn durations in milliseconds
+        FROM (
+            -- INNER QUERY: Consolidate each individual call before thread-level aggregation
+            -- This handles cases where calls_merged has multiple partial rows per call_id
+            -- due to ClickHouse materialized view background merge behavior
+            SELECT
+                id,                              -- Call identifier
+                any(thread_id) as thread_id,     -- Get any non-null thread_id for this call
+                                                -- (all non-null values should be identical)
+                min(started_at) as call_start_time,   -- Earliest start time for this call
+                max(ended_at) as call_end_time,   -- Latest end time for this call
+                -- Calculate call duration in milliseconds
+                CASE
+                    WHEN call_end_time IS NOT NULL AND call_start_time IS NOT NULL
+                    THEN dateDiff('millisecond', call_start_time, call_end_time)
+                    ELSE NULL
+                END as call_duration
+            FROM calls_merged
+            WHERE project_id = {pb_0: String}
+                AND sortable_datetime > {pb_1: String} AND sortable_datetime < {pb_2: String}
+            GROUP BY id                         -- Group by call id to merge partial rows
+            HAVING thread_id IS NOT NULL AND thread_id != '' AND id = any(turn_id) AND thread_id = {pb_3: String}  -- Filter to turn calls only
```

Comment:
if this HAVING clause gets large, as in the number of results that we have to group by here is large, we may want to include a WHERE optimization that filter down the calls as much as possible before grouping. Might be worth making a note here